### PR TITLE
✨ Limit inflight requests to the api-server

### PR DESCRIPTION
**Pull request by bisgaard-itis**
## What do these changes do?
- Introduce a new traefik middleware which limits the total number of in-flight requests to the api-server. This means there is an absolute limit to the number of requests the api-server will be handling at any given time (see https://doc.traefik.io/traefik/middlewares/http/inflightreq/). 
- If the number of allowed in-flight requests to the api-server is exceeded traefik returns a [429](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/429)
- Related ops MR: https://git.speag.com/oSparc/osparc-ops-deployment-configuration/-/merge_requests/651

## Related issue/s

- #5952 

## How to test

run local deployment with `API_SERVER_INFLIGHTREQ_AMOUNT=1` in `.env`. Then run
```bash
seq 10 | parallel -j 10 "curl -X 'GET'   'http://<ip address>.nip.io:<api-server-port>/v0/solvers'   -H 'accept: application/json'   -H 'Authorization: Basic <auth>' -v > output{#}.txt"
```
with `<api-server-port>`=the port exposing the api-server **through traefik** and `<auth>`=valid authorization.
Observe the `output*.txt` files and see that only a single request received a 2xx return code and the others received a 429.

## Dev-ops checklist

- [x] No ENV changes or I properly updated ENV ([read the instruction](https://git.speag.com/oSparc/osparc-ops-deployment-configuration/-/blob/configs/README.md?ref_type=heads#how-to-update-env-variables))

### CODE REVIEW

**Comment by YuryHrytsuk** [comment]
Thanks!

This limit will only take place for requests coming via `public` network. For the internal (within docker swarm cluster) communication it will not apply.

File: .env-devel [hunk 1]
```diff
@@ -21,6 +21,7 @@ AGENT_VOLUMES_CLEANUP_S3_SECRET_KEY=12345678
 API_SERVER_DEV_FEATURES_ENABLED=0
 API_SERVER_LOGLEVEL=WARNING
 API_SERVER_PROFILING=1
+API_SERVER_INFLIGHTREQ_AMOUNT=25
```
Comment (pcrespov) [inline .env-devel:24]: THOUGHT: I am afraid that at some point we might drop this env var if we do not see it in the api-server settings. I would add some reference to traefik in the name or prefix it as `TRAFIK_`

**Review by pcrespov — APPROVED** [review]
very nice!
Q: what is your strategy to determine this value (e.g. why did you start in 25  simultaneous requests?)

**Review by pcrespov — COMMENTED** [review]
Q: are you adding a sleep+retry in the client for this code?

**Comment by bisgaard-itis** [comment]
> very nice! Q: what is your strategy to determine this value (e.g. why did you start in 25 simultaneous requests?)

When running Werner's "load test" I see in our grafana dashboard that the api-server is at times handling around 50 requests simultaniously. That seems to add quite a performance overhead, so 25 is chosen to try and avoid that heavy load. Not sure this is optimal of course. So we will have to tune it as we go

File: .env-devel (hunk 1 above)
Comment (bisgaard-itis) [reply .env-devel:24]: Valid point. I will change.

**Review by sanderegg — APPROVED** [review]
All good! thanks!

**Comment by bisgaard-itis** [comment]
@YuryHrytsuk and @mrnicegyu11 can I get you guys (codeowners) to accept this one?

**Review by YuryHrytsuk — APPROVED** [review]

**Pull request merged**